### PR TITLE
Add version fields to RFQ wire messages

### PR DESCRIPTION
**Pull request by ffranr**
Closes https://github.com/lightninglabs/taproot-assets/issues/908

### CODE REVIEW

File: rfqmsg/reject.go [hunk 1]
```diff
@@ -102,9 +112,18 @@ var (
 	}
 )
 
+const (
+	// latestRejectVersion is the latest supported reject wire message data
+	// field version.
+	latestRejectVersion = V0
```
Comment (guggero) [inline rfqmsg/reject.go:118]: nit: shouldn't this just be "lastKnownVersion"? Since we won't reject V0.

Comment (guggero) [reply rfqmsg/reject.go:118]: Or just `latestVersion` similar to the name used in the next commit.

**Review by guggero — COMMENTED** [review]
Nice!

File: rfqmsg/buy_request_test.go [hunk 2]
```diff
@@ -46,6 +48,7 @@ func TestBuyRequestMsgDataEncodeDecode(t *testing.T) {
 	for _, tc := range testCases {
 		t.Run(tc.testName, func(tt *testing.T) {
 			req := buyRequestMsgData{
+				Version:       tc.version,
```
[the remark this answers is not in the document]
Comment (ffranr) [reply rfqmsg/buy_request_test.go:51]: In this PR, version checking happens after decoding (i.e. in `NewBuyRequestMsgFromWire`). So it shouldn't show up as an error in the encoding/decoding unit test. But I could add a non-zero value as version anyway here.

File: rfqmsg/reject.go (hunk 1 above)
Comment (ffranr) [reply rfqmsg/reject.go:118]: Here I attempted to use "latest version" constants which were specific to each message type. They are all in the same package so I think they need unique names.

Comment (guggero) [reply rfqmsg/reject.go:118]: Oh, I see... The word "reject" refers to the message type, not that we reject this version... Okay, never mind.

File: rfqmsg/buy_request_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (guggero) [reply rfqmsg/buy_request_test.go:60]: Yeah, that's okay. I mean that if we only encode the default value of 0, we don't actually test that any data is really encoded. So only when we add a non-zero value do we know encoding/decoding works on that field.

File: rfqmsg/buy_accept.go [hunk 3]
```diff
@@ -15,13 +15,23 @@ import (
...
-	TypeBuyAcceptAskPrice  tlv.Type = 2
-	TypeBuyAcceptExpiry    tlv.Type = 4
-	TypeBuyAcceptSignature tlv.Type = 6
-	TypeBuyAcceptAssetID   tlv.Type = 8
+	TypeBuyAcceptVersion   tlv.Type = 0
+	TypeBuyAcceptID        tlv.Type = 2
+	TypeBuyAcceptAskPrice  tlv.Type = 4
+	TypeBuyAcceptExpiry    tlv.Type = 6
+	TypeBuyAcceptSignature tlv.Type = 8
+	TypeBuyAcceptAssetID   tlv.Type = 10
```
Comment (Roasbeef) [inline rfqmsg/buy_accept.go:23]: Can we consolidate the buy/sell messages with what we had outlined here? https://github.com/Roasbeef/blips/pull/3

Comment (ffranr) [reply rfqmsg/buy_accept.go:23]: I'll spin that out into another PR. I think this PR is ready to be merged.

File: rfqmsg/messages.go [hunk 4]
```diff
@@ -115,6 +116,45 @@ func NewIncomingMsgFromWire(wireMsg WireMessage) (IncomingMsg, error) {
...
+	// V0 represents version 0 of the contents in a wire message data field.
+	V0 WireMsgDataVersion = 0
+)
+
+// WireMsgDataVersionEncoder is a function that can be used to encode a
+// WireMsgDataVersion to a writer.
+func WireMsgDataVersionEncoder(w io.Writer, val any, buf *[8]byte) error {
+	if version, ok := val.(*WireMsgDataVersion); ok {
+		versionUint8 := uint8(*version)
+		return tlv.EUint8(w, &versionUint8, buf)
```
Comment (jharveyb) [inline rfqmsg/messages.go:133]: nit: why not `EUint8T` like in `asset.VersionEncoder`?

```go
return tlv.EUint8T(w, uint8(*version), buf)
```

**Review by jharveyb — APPROVED** [review]
LGTM 💯

**Review by guggero — APPROVED** [review]

File: rfqmsg/messages.go (hunk 4 above)
Comment (ffranr) [reply rfqmsg/messages.go:133]: It should have been `tlv.EUint8T`. I'll try to keep that in mind in the future. I didn't get a chance to make that modification in this PR.

Comment (jharveyb) [reply rfqmsg/messages.go:133]: Sg, could be in the follow-up PR you mentioned above.

**Pull request merged**